pkg/provisioner/ironic/ironic.go

```diff
@@ -673,13 +675,27 @@ func (p *ironicProvisioner) getUpdateOptsForNode(ironicNode *nodes.Node) (update
 		op = nodes.ReplaceOp
 		p.log.Info("updating root_device")
 	}
-	hints := map[string]string{}
-	switch {
-	case hwProf.RootDeviceHints.DeviceName != "":
-		hints["name"] = hwProf.RootDeviceHints.DeviceName
-	case hwProf.RootDeviceHints.HCTL != "":
-		hints["hctl"] = hwProf.RootDeviceHints.HCTL
+
+	// hints
+	//
+	// If the user has provided explicit root device hints, they take
+	// precedence. Otherwise use the values from the hardware profile.
+	//
+	// NOTE: This lookup logic is reproduced in the controller's
+	// actionProvisioning() method where the values are stored in the
+	// status struct. There wasn't a good place to put a common method
+	// to do the lookup, and it's a temporary situation until we
+	// remove hardware profiles, so this seemed like a reasonable
+	// approach.
```

Comment:
Surely we should use the value we copied to the Status, not the one from the Spec, anyway? That way there's no possibility that what we've stored in the Status to tell the user which values we used at the time of provisioning could be different from which values we actually used at the time of provisioning.